syntax = "proto3";

package embedding;

option go_package = "github.com/dremotha/mapbot/api/proto/embedding";

service EmbeddingService {
    rpc Embed(EmbedRequest) returns (EmbedResponse);
    rpc EmbedBatch(EmbedBatchRequest) returns (EmbedBatchResponse);
    rpc Health(HealthRequest) returns (HealthResponse);
}

message EmbedRequest {
    string text = 1;
}

message EmbedResponse {
    repeated float vector = 1;
}

message EmbedBatchRequest {
    repeated string texts = 1;
}

message EmbedBatchResponse {
    repeated EmbedResponse vectors = 1;
}

message HealthRequest {}

message HealthResponse {
    bool healthy = 1;
    string model_name = 2;
    int32 vector_size = 3;
}




